### ISSUE TITLE
Breaking changes in @azure/msal-node 2.6.2

### ISSUE THREAD

**Issue by maorleger** [AzureAD/microsoft-authentication-library-for-js#6859]
### Core Library

MSAL Node (@azure/msal-node)

### Core Library Version

2.6.2

### Wrapper Library

Not Applicable

### Wrapper Library Version

None

### Public or Confidential Client?

Confidential

### Description

Looks like a change was released in 2.6.2 that changes the type of exceptions thrown from MSAL on various scenarios. There are a few examples, but I created a minimal repro showing a specific error that has changed.

While I do agree that the new error message is clearer, it _is_ a breaking change for customers who rely on the endpoints_resolution_error message and ClientAuthError type, like @azure/identity does.

I created a minimal repro showing how the exception differs in @msal/msal-node 2.6.1 vs 2.6.2 when passing an invalid tenant ID but I want to highlight that this is _not_ the only breaking change in the exception flows so I would recommend using it as an example of a wider breaking change

### Network Trace (Preferrably Fiddler)

- [ ] Sent
- [ ] Pending

### MSAL Configuration

```javascript
{
    auth: {
      clientId: "clientId",
      // Specify an invalid tenant id to trigger the error
      authority: "https://login.microsoftonline.com/invalid-tenant-id",
      clientSecret: "clientSecret",
    }
}
```

### Relevant Code Snippets

```javascript
// Please see https://github.com/maorleger/msal-2.6.2-repro for how I configured 2.6.1 and 2.6.2 side-by-side

const msal261 = require("@azure/msal-node-2.6.1");
const msal262 = require("@azure/msal-node-2.6.2");

async function main(msal) {
  console.log("using version: ", msal.version);
  const config = {
    auth: {
      clientId: "clientId",
      // Specify an invalid tenant id to trigger the error
      authority: "https://login.microsoftonline.com/invalid-tenant-id",
      clientSecret: "clientSecret",
    },
  };

  const cca = new msal.ConfidentialClientApplication(config);

  const tokenRequest = {
    // scopes doesn't matter in this case, just needs to be non-empty
    scopes: ["https://graph.microsoft.com/.default"],
  };

  const response = await cca.acquireTokenByClientCredential(tokenRequest);
  console.log(response);
}

main(msal261)
  .catch(console.error)
  .then(() => main(msal262))
  .catch(console.error);
```

### Reproduction Steps

https://github.com/maorleger/msal-2.6.2-repro creates a minimal repro showing the side-by-side behavior of the two versions.

### Expected Behavior

As a patch version bump I would expect no behavior changes

### Identity Provider

Entra ID (formerly Azure AD) / MSA

### Browsers Affected (Select all that apply)

None (Server)

### Regression

@azure/msal-node 2.6.1

### Source

Internal (Microsoft)

**Comment by tnorling** [comment]
This is not a change to the API surface but rather a completely different error. In 2.6.2 we fixed a bug affecting how we resolve endpoints. MSAL.js ships with hardcoded endpoints for Microsoft authorities and the bug prevented these from being used, resulting in making a network request to resolve these endpoints.

2.6.1 - Resolve endpoints over network, request is failing with endpoints_resolution_error
2.6.2 - Resolve endpoints via hardcoded metadata, then attempt to make a request and server fails with invalid tenant

TLDR: This isn't a change from one error to the other, in the new version you're getting one step further than you were before but you still have incorrect config.

**Comment by tnorling** [comment]
Just to add: you can replicate both errors in both versions

2.6.1 - Pass endpoint metadata on cloudDiscoveryMetadata and authorityMetadata - you'll get the same error you're getting in 2.6.2
2.6.2 - Pass an invalid non-Microsoft authority - you'll get endpoints_resolution_error

That said I wouldn't recommend taking a hard dependency on specific errors, like it sounds like you are.

**Comment by maorleger** [comment]
Appreciate the quick reply. I get what you're saying and get that this is a case of [Hyrum's Law](https://www.hyrumslaw.com/) in action. I think this may have a very real impact on @azure/identity customers since Identity relies on the error codes to determine what exception to throw (you can see that code [here](https://github.com/Azure/azure-sdk-for-js/blob/c7301ab78063fa366ec6d08254e5c02806e8be30/sdk/identity/identity/src/msal/utils.ts#L147-L195))

I can point them to this thread if it comes up just so we can have a good idea of impact. So far it's just impacting Identity tests which I can fix up.

> That said I wouldn't recommend taking a hard dependency on specific errors, like it sounds like you are.

Agreed, but unfortunately that code has been around for quite some time and is in use by the Azure Identity library today. I can figure out how we may make it more robust.

**Comment by tnorling** [comment]
You're missing error handling for ServerErrors. Not sure if that's intentional or not. I get what you're saying but at least in this case this is a result of a configuration error, meaning the only thing that might want to take dependency on it is tests. Any working app should not be running into this.
